src/foam/nanos/crunch/UserCapabilityJunctionDAO.js

```diff
@@ -126,35 +146,76 @@ foam.CLASS({
           name: 'x',
           type: 'Context'
         },
+        {
+          name: 'capability',
+          type: 'Capability'
+        },
         {
           name: 'obj',
           type: 'foam.nanos.crunch.UserCapabilityJunction'
+        },
+        {
+          name: 'putObject',
+          type: 'Boolean'
         }
       ],
+      javaType: 'foam.core.FObject',
       documentation: `
       We may or may not want to store the data in its own dao, based on the nature of the data.
       For example, if the data for some UserCapabilityJunction is a businessOnboarding object, we may want to store this object in
       the businessOnboardingDAO for easier access.
       If the data on an UserCapabilityJunction should be stored in some DAO, the daoKey should be provided on its corresponding Capability object.
       `,
       javaCode: `
-      DAO capabilityDAO = (DAO) x.get("capabilityDAO");
-      Capability capability = (Capability) capabilityDAO.find(obj.getTargetId());
-
+      if ( obj.getData() == null ) 
+        throw new RuntimeException("UserCapabilityJunction data not submitted for capability: " + obj.getTargetId());
+      
       String daoKey = capability.getDaoKey();
-      if ( daoKey == null ) return;
+      if ( daoKey == null ) return null;
 
       DAO dao = (DAO) x.get(daoKey);
-      if ( dao == null ) return;
+      if ( dao == null ) return null;
+
+      FObject objectToSave;                                                  // Identify or create data to go into dao.
+      String contextDAOFindKey = (String)capability.getContextDAOFindKey();
+
+      if ( contextDAOFindKey != null && ! contextDAOFindKey.isEmpty() ) {
+        if ( contextDAOFindKey.toLowerCase().contains("subject") ) {         // 1- Case if subject lookup
```

Comment:
What is this for?

Reply:
The contextDAOFindKey specifies the object in the context to update. 
The format we are using for users in the system, with the new Subject is:
"contextDAOFindKey": "subject.user"

then in this decorator.saveDataToDAO, we know to look up the subject in the context, and use getUser() or getRealUser() to update the correct context object